### PR TITLE
Add components folder and Api.js to repo

### PR DESCRIPTION
**Pull request by michaellawes**
I wanted to rename the containers folder I made to components for the React Components but forgot to push the folders back into the repo

### CODE REVIEW

**Review by MufaroMakiwa — APPROVED** [review]

File: src/components/Auth/Auth.js [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+  };
+
+  // Validates user entered sign up information correctly
+  const handleOnSignUpSubmit = (username, password, email, callBack) => {
+    let error = false;
+    if (username === '') {
+      setUsernameError(true);
+      setUsernameErrorMsg('Username cannot be empty');
+      error = true;
+    } else setUsernameError(false);
```
Comment (andiov) [inline src/components/Auth/Auth.js]: Nit, I'd keep the bracket style consistent in all files. This applies everywhere you use else in this file.

```
if {
  ...
} else {
  ...
}
```

**Review by andiov — COMMENTED** [review]
Check all the files you committed here. You have a couple that end in ~. Those should not be submitted. Those are backup files and they're automatically generated, I think it might be related to the editor you use. But you should delete them from your repo.

File: src/components/Auth/Auth.js (hunk 1 above)
Comment (michaellawes) [reply src/components/Auth/Auth.js]: Fixed along with removing the backup files from the commit

Comment (andiov) [reply src/components/Auth/Auth.js]: I think when you made edits, you added back the backup ~ files. Please remove them before submitting.

Also, not seeing the else changes. Basically this is what I was looking for:
```
   if (username === '') {
      setUsernameError(true);
      setUsernameErrorMsg('Username cannot be empty');
      error = true;
    } else {
      setUsernameError(false);
   };
```

Comment (michaellawes) [reply src/components/Auth/Auth.js]: The changes are in new commits specifically eb392cb and 6ee58f6

**Review by andiov — APPROVED** [review]

**Pull request merged**